### ISSUE TITLE
Taint Analysis: `where()` whole-argument and flat-map strip paths lack the guards the element-wise path has (false negatives)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1306]
Surfaced by an external review of #1305 and verified against source. Both are **false negatives**: SQL taint is stripped where Laravel would still emit the value raw, so a genuine injection goes unreported.

#1302 gave the element-wise strip path two guards that the other two strip paths never received.

### 1. The whole-argument strip is not scoped to a Laravel builder

`beforeExpressionAnalysis()` records the first argument for **any** receiver:

```php
self::$whereColumnArgumentIds[\spl_object_id($argument)] = true;   // WhereColumnTaintHandler.php:221
```

Element-wise removal then checks the receiver:

```php
if (!self::isLaravelBuilder($position['receiver'], $statements_source, $event)) {   // :291
    return 0;
}
```

`removeWholeArgumentTaints()` (:255-271) checks only argument-id membership and `isBoundValueMap($type)` — no receiver check. So a project's own non-builder `where()` loses the taint:

```php
final class ReportQuery
{
    public function where(array $parts): void
    {
        DB::unprepared($parts['sql']);          // injection, now unreported
    }
}

(new ReportQuery())->where(['sql' => (string) $request->term]);
```

The comment at :223-229 already spells out this exact risk ("a project's own `ReportQuery::where(array $parts)` that interpolates `$parts[0]` into raw SQL would otherwise have its list element stripped") and stores the receiver for the element-wise path only. The whole-argument path predates #1302 (the `isBoundValueMap` type-level path from the #734 / #1218 lineage), so this is not a #1302 regression — it is the older path never getting the same treatment.

Fix: store and verify the receiver for whole-argument records too.

### 2. Keyed-map values are stripped even when Laravel emits them raw

For a non-numeric string key, `scalar_gated` is false (:376), making removal unconditional (:295); the non-literal path likewise examines only keys (:523). But `addBinding()` is skipped for an `ExpressionContract`, whose `getValue()` the grammar emits verbatim. The nested path already recognises precisely this and keeps the sink (:420); the flat and whole-map paths need an equivalent raw-capable-object gate.

Reachable via `where(['name' => $expr])` where `$expr` is a user-influenced `ExpressionContract`. Narrower than item 1, but the same shape of hole.

### Scope

- Add the receiver check to the whole-argument path.
- Add a raw-capable-object gate to the flat and whole-map paths, mirroring :420.
- phpt coverage for both, using a non-builder receiver and an `ExpressionContract` value respectively. Mind the batch-swallow artifact when picking sink methods (see `docs/contributing/taint-analysis.md`).
